go/private/rules/transition.bzl

```diff
@@ -44,21 +40,16 @@ load(
 def filter_transition_label(label):
     """Transforms transition labels for the current workspace.
 
-    This is a workaround for bazelbuild/bazel#10499. If a transition refers to
-    a build setting in the same workspace, for example
-    @io_bazel_rules_go//go/config:goos, it must use a label without a workspace
-    name if and only if the workspace is the main workspace.
-
-    All Go build settings and transitions are in io_bazel_rules_go. So if
-    io_bazel_rules_go is the main workspace (for development and testing),
-    go_transition must use a label like //go/config:goos. If io_bazel_rules_go
-    is not the main workspace (almost always), go_transition must use a label
-    like @io_bazel_rules_go//go/config:goos.
+    This works around bazelbuild/bazel#10499 by automatically using the correct
+    way to refer to this repository (@io_bazel_rules_go from another workspace,
+    but only repo-relative labels if this repository is the main workspace).
     """
-    if IS_RULES_GO and label.startswith("@io_bazel_rules_go"):
-        return label[len("@io_bazel_rules_go"):]
-    else:
+    if label.startswith("//command_line_option:"):
+        # This is a special label used to refer to Bazel command-line options in
```

Comment:
Can you add a note about why are these special and when it could be removed? My experience is that  hacks like this get put in place and I forget to write down the condition where it can be removed safely so I carry forward scar tissue.

Reply:
Good point, I expanded the comment. Could you give it another read?